makeservices/easywp

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Creates symlink to user's webroot. If the symlink exists and files
+# are present in webroot, they are backed up in the user's home directory.
+# A user's database is created or, if it already exists, the password 
+# is reset. WordPress is installed and the wp-config file is filled
+# out with the requisite information. Further configuration can be 
+# done by the user from a browser.  
+#
+# WARNING: This script should generally be run on a account without
+#          an existing website or database!
+#
+
+set -euxo pipefail
+shopt -s dotglob
+
+php_defines() {
+    sed -i "s/$1/$2/" "$WPCONFIG"
+}
+
+WEBROOT=$HOME/public_html
+WPCONFIG=wp-config.php
+USER=$(whoami)
+
+# Create symlink to webroot
+makehttp
+cd "$WEBROOT"
+
+# If user has files, move them to a new location. 
+if [ -n "$(ls -A)" ]; then
+    TS=$(date +"%Y-%m-%d_%H:%M:%S")
+    BACKUP="$HOME/webroot-$TS"
+    mkdir -p "$BACKUP"
+    echo "Webroot not empty. Moving current files to $BACKUP..." 
+    mv ./* "$BACKUP"
+fi
+
+# Get SQL password
+echo "Resetting database..."
+SQLPASS=$(echo "yes" | makemysql | tail -n 1 | grep -Po '(?<=: )([0-9a-zA-Z]){24,}$')
+if [ -z "$SQLPASS" ]; then
+    echo "  Error: Could not retrieve database password. Run makemysql to see the issue."
+    exit 1
+fi
+echo "SQL set up and the password can be found in $WPCONFIG ..."
+
+# Install WordPress
+wp core download
+cp wp-config-sample.php "$WPCONFIG"
+
+# dos2unix
+sed -i 's/\r//' "$WPCONFIG"
+
+# Modify the config file with correct info
+php_defines 'database_name_here' "$USER"
+php_defines 'username_here' "$USER"
+php_defines 'password_here' "$SQLPASS"
+php_defines 'localhost' 'mysql'
+php_defines 'utf8' 'utf8mb4'
```

Comment:
It just occurred to me---couldn't you do `wp config set ...` here? Would it make the script simpler/easier to understand?

Reply:
Oh rip... ```wp-config``` also grabs the salts too. Had no idea that existed. Yeah, that'll definitely simplify things.